pinot-common/src/main/java/org/apache/pinot/sql/parsers/rewriter/NonAggregationGroupByToDistinctQueryRewriter.java

```diff
@@ -85,9 +85,7 @@ public PinotQuery rewrite(PinotQuery pinotQuery) {
       pinotQuery.setGroupByList(null);
       return pinotQuery;
     } else {
-      throw new SqlCompilationException(String.format(
-          "For non-aggregation group-by query, select expression set and group-by expression set should be the same. "
-              + "Found select: %s, group-by: %s", selectExpressions, groupByExpressions));
+      return pinotQuery;
```

Comment:
@tarun11Mavani : what is the behavior for queries that reference a column not in the grouping set? Say

```
SELECT someCol FROM table GROUP BY col1, col2
```

Ideally we should fail such queries in the broker itself.

Reply:
This query will fail in such cases due to a validation process that occurs after all rewrites are completed in the [CalciteSqlParser](https://github.com/apache/pinot/blob/53fbf88027c47b3c6f7d1526576ba0bd257fe9d5/pinot-common/src/main/java/org/apache/pinot/sql/parsers/CalciteSqlParser.java#L550). Specifically, the [validateGroupByClause](https://github.com/apache/pinot/blob/53fbf88027c47b3c6f7d1526576ba0bd257fe9d5/pinot-common/src/main/java/org/apache/pinot/sql/parsers/CalciteSqlParser.java#L550) function is invoked to recursively check whether an expression contains any references that are not included in the GROUP BY clause. As a result, there is no need to perform this verification in the NonAggregationGroupByToDistinctQueryRewriter.